CDCgov/prime-reportstream
# Send Failure Log Enhancement

This PR adds the associated submitted IDs in the "Send Failed" log message when a report fails to send to a receiver. This is immediately expected to help Flexion engineers quickly trouble-shoot issues with NBS.

**Test Steps (local):**
1. Stop the docker container running SFTP
2. Submit a report that is destined to a receiver with the default SFTP setup
3. Verify the log message includes the associated submitted report(s):

```
[2024-10-11T20:36:03.756Z] {"mdc":{"receiver":"ignore.ETOR_TI","reportId":"4a222994-233d-48eb-ac81-35faa08625e8"},"message":"Send Failed.  Will retry sending report: ReportFile (4a222994-233d-48eb-ac81-35faa08625e8, 293, send, null, null, null, ignore, ETOR_TI, null, null, None, ETOR_TI, http://localhost:10000/devstoreaccount1/reports/ready%2Fignore.ETOR_TI%2F4a222994-233d-48eb-ac81-35faa08625e8.hl7, null, HL7, [binary...], 1, null, 2024-10-11T15:36:01.398757-05:00, null, 1).reportId to ignore.ETOR_TI in 10 minutes and -16 seconds at 2024-10-11T15:45:47.745888-05:00. Corresponding submitted ReportIds: [1f55b381-b11e-455f-b254-c91e7e9c35d9]","thread":"pool-2-thread-66","timestamp":"2024-10-11T20:36:03.756Z","level":"WARN","logger":"gov.cdc.prime.router.azure.SendFunction"}
```

## Changes
- SendFunction now includes the associated submitted report IDs in the "Send Failed" log message 

### Testing
